testcert.py

```diff
@@ -108,83 +128,164 @@ def _ParseArgs():
     if 'ty' in v['info'].properties:
       if options.printer in v['info'].properties['ty']:
-        pinfo = str(v['info']).split(',') 
+        pinfo = str(v['info']).split(',')
         for item in pinfo:
           if 'port' in item:
             privet_port = int(item.split('=')[1])
-            logger.debug('Privet advertises port: %d', privet_port)
-  device = Device(logger, chromedriver, privet_port=privet_port)
-  transport = Transport(logger)
-  time.sleep(2)
+            _logger.debug('Privet advertises port: %d', privet_port)
+
+  _gcp = GCPService(Constants.AUTH["ACCESS"])
+  _device = Device(_logger, Constants.AUTH["ACCESS"], _gcp, privet_port=privet_port)
+  _transport = Transport(_logger)
 
   if Constants.TEST['SPREADSHEET']:
-    global sheet
-    sheet = _sheets.SheetMgr(logger, chromedriver, Constants)
-    sheet.MakeHeaders()
+    global _sheet
+    _sheet = _sheets.SheetMgr(_logger, _storage.get(), Constants)
+    _sheet.MakeHeaders()
   # pylint: enable=global-variable-undefined
 
 
-def tearDownModule():
-  chromedriver.CloseChrome()
+def LogTestSuite(name):
+  """Log a test result.
+
+  Args:
+    name: string, name of the testsuite that is logging.
+  """
+  print '=============================================================================================================='
+  print '                                     Starting %s testSuite'% (name)
+  print '=============================================================================================================='
+  if Constants.TEST['SPREADSHEET']:
+    row = [name,'','','','','','','']
+    _sheet.AddRow(row)
+
+
+def waitForPrivetDiscovery(printer, browser):
+  t_end = time.time() + 30
+
+  while time.time() < t_end:
+    for v in browser.listener.discovered.values():
+      if 'info' in v:
+        if 'ty' in v['info'].properties:
+          if printer in v['info'].properties['ty']:
+            if v['found']:
+              return True
+    Sleep('POLL')
+  # Timed out
+  return False
+
+
+def isPrinterAdvertisingAsRegistered(printer):
+  """Checks the printer's privet advertisements and see if it is advertising as registered or not
+
+      Args:
+        printer: string, printer name
+      Returns:
+        boolean, True = advertising as registered, False = advertising as unregistered, None = advertisement not found
+      """
+  for v in _mdns_browser.listener.discovered.values():
+    if 'info' in v:
+      if 'ty' in v['info'].properties:
+        if printer in v['info'].properties['ty']:
+          properties = v['info'].properties
+          return properties['id'] and 'online' in properties['cs'].lower()
+  return None
+
+def waitForAdvertisementRegistrationStatus(name, is_wait_for_reg, timeout):
+  """Wait for the device to privet advertise as registered or unregistered
+
+      Args:
+        name: string, device status to wait on
+        is_wait_for_reg: boolean, True for registered status, False for unregistered
```

Comment:
I agree that an Enum would make the calling code a lot clearer. But I like the fact that a boolean forces the input to be binary - either true or false; whereas I would have to potentially handle unexpected Enum inputs from the caller if for whatever reason the another field is added to the Enum in the future. Also, I would either have to create a module level Enum in testcert.py or jam it into _config.py, both solutions aren't appealing to me. Thoughts?